app/services/call_tool/call_status.rb

```diff
@@ -0,0 +1,18 @@
+module CallTool
+  module CallStatus
+    TIME_OUT_THRESHOLD = 15.seconds
+
+    # Returns one of: connecting, ringing, answered, completed, timed_out
+    def self.for(call)
+      status = call.member_call_events.last&.dig('CallStatus')
```

Comment:
All very good points. The trick about `dig` is getting nested keys, so it doesn't make sense to use it here really. Ideally i'd like to call `[]` using the safe navigation operator, but I couldn't figure out how (not sure it's possible). `fetch` on the other hand, raises an error if the key doesn't exist, and since I'm not doing any validations on the status hashes, I don't think that's a good idea. So I was left with `dig`, it's not made to be used like this but it's the best way I figured out how to do it.